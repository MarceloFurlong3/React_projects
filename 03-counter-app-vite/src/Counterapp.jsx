import {useState} from 'react'
import PropTypes from 'prop-types';


export const Counterapp = ({ value }) => {
    
    const [counter,setCounter] = useState(value)
    
    const handleadd = () => {

      //setCounter( counter + 1 );

        setCounter((c) =>  c + 1)
    }
    
    const takeout = () => { setCounter(counter - 1)}
    
    const reset = () => { setCounter(() => value) }

   return (
    <>
    
    <h1>CounterApp</h1>
    
    <h2> { counter } </h2>

    <button onClick={ handleadd } > 
        +1
    </button>

    <button onClick={takeout}> 
        -1 
    </button>

    <button aria-label="btn-reset" onClick={reset}>
        reset 
    </button>

    </>
);
}

Counterapp.protoTypes = {
    value:PropTypes.number.isRequired
}


//rafc tab creamos un funcional component