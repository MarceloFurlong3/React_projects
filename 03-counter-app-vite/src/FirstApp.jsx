import PropTypes from 'prop-types';


export const FirstApp = ({
    title = "hola soy un perro",
    subTitle =  "soy un substitulo",
    name
}) => {



    return (
        <>
        
        <h1 data-testid="test-title">{ title }</h1>
        
        <p> {subTitle} </p>
        <p> {subTitle} </p>
        <p> {name} </p>
        
        </>
    )
}

FirstApp.propTypes = {
title:PropTypes.string.isRequired,
subTitle:PropTypes.string //si ponenemos .requaried entonces no podemos usar default si o si tenemso que darle valor
}


FirstApp.defaultProps = {
    //title:"no hay titulo",
    subTitle:"no hay substitulo",
    name:"soy un cacahuate"
}