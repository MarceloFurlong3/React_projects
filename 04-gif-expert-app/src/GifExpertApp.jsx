import { useState } from "react";
import { AddCategory } from "./components/AddCategory";


export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch','dragon ball'])

  const onAddCategory = (newCategory) => {
   if (categories.includes(newCategory)) return ;

     setCategories([newCategory,...categories])

  }


return (

    <>

        {/* titulo */}
        <h1>GifExpertApp</h1>

        {/* input */}
        <AddCategory
            onNewCategory={onAddCategory}>

        </AddCategory>

        {/* listado de cards */}
        
        <ol>
            {categories.map(category => {

                return <li key={category}>{category}</li>
            })}
    
        </ol>


        {/* Gif Item */}



     </>


    )


}