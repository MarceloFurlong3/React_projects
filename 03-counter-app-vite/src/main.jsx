import React from 'react';
import ReactDOM from 'react-dom/client';
import { Counterapp } from './Counterapp';
import { FirstApp } from './FirstApp';
import { Marcelo } from './NameApp';
import './stles.css'

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <Counterapp value={1}></Counterapp>
        {/* <Marcelo></Marcelo> */}
    {/* <FirstApp title='Titulazo'></FirstApp> */}
    </React.StrictMode>

)


