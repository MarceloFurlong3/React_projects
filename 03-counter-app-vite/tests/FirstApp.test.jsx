import { render,screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('Pruebas en <fIRSTAPP></fIRSTAPP>', () => { 

  const title = "hola ,soy goku";
  const subTitle = "soy un substitulo"


    test(' debe hacer match con el snapshot',() => {

      const {container} = render (<FirstApp title={title}></FirstApp>)
      expect(container).toMatchSnapshot();
    })

    test('debe de moestrar el mensaje "hola,soy goku"', () => { 

       render(<FirstApp title={title}></FirstApp>)
      
       expect(screen.getByText(title)).toBeTruthy();

      })

    test('debe de moestrar el titulo en un h1' ,() => {

      render(<FirstApp title={title}></FirstApp>)

      expect(screen.getByRole('heading',{level : 1}).innerHTML).toContain(title);


    });


    test('debe de moestrar el substitulo enviado por props', () => { 



      render(
      <FirstApp 
    
        title={title}
        subTitle={subTitle}

      ></FirstApp>)

      expect(screen.getAllByText(subTitle).length).toBe(2);
        



     })



 });