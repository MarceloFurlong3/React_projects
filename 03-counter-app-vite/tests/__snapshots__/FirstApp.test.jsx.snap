// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <fIRSTAPP></fIRSTAPP>  debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    hola ,soy goku
  </h1>
  <p>
     
    no hay substitulo
     
  </p>
  <p>
     
    no hay substitulo
     
  </p>
  <p>
     
    soy un cacahuate
     
  </p>
</div>
`;
